### ISSUE TITLE
make GUI Box Properties call 'apply' on choosing the Font Size + 'enter' inconsistency + other font sizes

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#2484]
The radio buttons for the label position calls the 'apply' function in properties, so we see the changes while we're setting it and choosing it. On the other hand, the 'Font Size' settings don't and you need to hit the 'ok' button, but I find it even more useful in this case.

I also find something weird and inconsistent, which is how the enter key works. For the boxes that take numbers and symbols, when you hit enter it goes to the 'ok' button and things get applied, but if you select the same box or another one instead, the focus from the 'ok' button is removed. In the case of the radio buttons this is not the case. You can hit enter and it'll focus the 'ok' button, but if you choose another radio button setting the 'ok' button keeps highlighted.

This is particularly annoying in the case of GUI Boxes becasue if I want to test different sizes I need to first hit enter, so it gets applyed and the 'ok' button is focused, but if I want to test another size setting while clicking on the radio button the size doesn't change and the 'ok' button is not deselected, so if I hit ok the properties get closed because it's like hitting the 'ok' button for good.

While we're at it, I'd also like to have a font size box to enter another custom value, so we can have a box to enter such values and instead of the radio buttons we could just have something like the sample rate settings, with buttons for most popular settings, but the possibility to set any other size. For the record, this is also discussed in https://github.com/pure-data/pure-data/issues/1344 see also https://github.com/pure-data/pure-data/issues/1836 which is related as well

**Comment by umlaeute** [comment]
afaict, this is only a macOS issue, as this is the only platform where changes will be immediately apply.
on the other platforms (Windows, Linux) you *always* have to approve a change with `Apply` (or `OK`) first, before you see them.

---
and I'm not sure whether you are aware, that having multiple different feature requests in a single github issue, makes it somewhat hard to action upon (when do we close it?). if you have multiple issues, please create multiple tickets (even if they all are annoying you simulatenously in a particular situation)

**Comment by porres** [comment]
yes, this feels like the case where I can split issues. Let me think about it and reorganize it. But I also pointed to related discussions in other issues, so it feels already complex and convoluted as it is.

------------------

> afaict, this is only a macOS issue, as this is the only platform where changes will be immediately apply.
> on the other platforms (Windows, Linux) you always have to approve a change with Apply (or OK) first, before you see them.

Yeah, I see, macOS only has "cancel" and "ok" buttons, right? In the others you have to hit "apply". I think I remember a discussion about this before. This is done because this is the standard for macOS or something, right?

Well, ok, so the issue is for macOS only

**Comment by umlaeute** [comment]
closing this, as there are too many different issues.

if you decide to report them one by one, i would highly recommend to be specific about which objects you are talking.

e.g. 
> radio buttons for the label position calls the 'apply' function in properties

hints that you are talking about the gatom properties (as there are no radio boxes for label position with the iemguis).

as for:
> but if you select the same box or another one instead

i have really no clear idea what you mean with "box" here.

probably illustrate the issue you are having with screenshots.

**Comment by porres** [comment]
ok
